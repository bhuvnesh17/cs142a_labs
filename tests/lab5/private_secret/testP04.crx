// test if-else and while checks for boolean
func compare_if(a : int) : int {
  if (a) {
    ::printInt(7);
  }
  ::printInt(5);
  return 0;
}

func compare_while(a:int) : int {
  while (a) {
    ::printInt(7);
    let a = a - 1;
  }
  return 0;
}

func main() : void {
  ::compare_if(4);
  ::compare_while(8);
}
