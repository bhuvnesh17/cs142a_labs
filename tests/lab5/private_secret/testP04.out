Error type-checking file.
TypeError(3,3)[IfElseBranch requires bool condition not int.]
TypeError(11,3)[WhileLoop requires bool condition not int.]

